File: C++/Kelas_Terbuka/Materi/operatorlogika.cpp
#include <iostream>
using namespace std; 
int main() {
    int a = 2;
    int b = 3;
    bool hasil; 

    // Belajar operator logika : not, and, or


    // Operator not(!)
    hasil = !(a == 3);

    // Operator and, kedua nilai harus true agar menghasilkan satu, jika salah satunya ada yang false, maka dia menghasilkan nol(0). 
    hasil = ( a == 3 ); 
    
    hasil = (a == 3) and (b == 2); // true and true
    cout << "hasilnya adalah: " << hasil << endl; 
    hasil = (a == 4) and (b == 2); // false and true
    cout << "hasilnya adalah: " << hasil << endl;
    hasil = (a == 3) and (b == 3); // true and false
    cout << "hasilnya adalah: " << hasil << endl;
    hasil = (a == 4) and (b == 3); // false and false  
    cout << "hasilnya adalah: " << hasil << endl;

    // Operator or, jika salah satu bernilai true, maka hasilnya akan true (1)
    hasil = (a == 3) or (b == 2); // true and true
    cout << "hasilnya adalah: " << hasil << endl; 
    hasil = (a == 4) or (b == 2); // false and true
    cout << "hasilnya adalah: " << hasil << endl;
    hasil = (a == 3) or (b == 3); // true and false
    cout << "hasilnya adalah: " << hasil << endl;
    hasil = (a == 4) or (b == 3); // false and false  
    cout << "hasilnya adalah: " << hasil << endl;

    cin.get(); 
    return 0; 
}

File: Javascript_Course/Javascript_PZN/Javascript_Dasar/operator/operator_logika.js
// Operator logika adalah operator untuk dua buah data boolean. Hasil dari operator logika adalah boolean lagi
// Contoh proses penentuan piala AFF Final:

let Indonesia;
let Malaysia;

Indonesia = 5;
Malaysia = 5;

const injuryTime = 10;

Indonesia = 6;
Malaysia = 6;

if (Indonesia == Malaysia) {
  console.log("Pertandingan akan dilanjutkan dengan penalti. ");
}

else {
  console.log("Salah satu pihak menang");
}

// Contoh pemakaian operator unary (!) atau kebalikan nilai

let mangga = 100;
let jeruk = 200;

if (mangga < jeruk) {
  console.log(!false); // Walau nilainya false, tapi karena di depannya ada operator unary, maka hasilnya dibalikin jadi true
}
